uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php

```diff
@@ -540,4 +541,45 @@ public static function getOneWithMustTakePublicHolidayAsLeaveRequest() {
 
     return null;
   }
+
+  /**
+   * Returns the date the TOIL will expire based on the date given to the method.
+   *
+   * @param \DateTime $date
+   *   The date to calculate TOIL expiry date
+   *
+   * @return \DateTime|null
+   *   returns null when TOIL never expires for the Absence Type
+   *
+   * @throws CRM_HRLeaveAndAbsences_Exception_InvalidAbsenceTypeException
+   */
+  public function calculateToilExpiryDate(DateTime $date) {
+
+    if (!$this->allow_accruals_request) {
+      throw new InvalidAbsenceTypeException("This Absence Type does not allow Accruals Request");
+    }
+    if ($this->toilNeverExpires()) {
+      return null;
+    }
+
+    if ($this->accrual_expiration_unit == self::EXPIRATION_UNIT_DAYS) {
+      $unit = 'day';
+    }
+    if ($this->accrual_expiration_unit == self::EXPIRATION_UNIT_MONTHS) {
+      $unit = 'month';
+    }
+
+    $date->modify('+'.$this->accrual_expiration_duration. ' ' . $unit);
+
+    return $date;
+  }
+
+  /**
+   * This method determines whether the Absence Type TOIL expires or not
+   *
+   * @return bool
+   */
+  public function toilNeverExpires() {
```

Comment:
If the method is public, you need to add a test to it. Another option is to make the method private, then you won't need the test. I prefer the second option, since it keeps the public interface of the class as small as possible. 

In the future, if we need to use the method outside the class, we can make it public.